prefect2_api/block_document.go

```diff
@@ -0,0 +1,121 @@
+package prefect2_api
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+func (c *Client) GetAllBlockDocuments(workspaceID string) ([]BlockDocument, error) {
+	req, err := http.NewRequest("POST", fmt.Sprintf("%s/accounts/%s/workspaces/%s/block_documents/filter", c.PrefectApiUrl, c.PrefectAccountId, workspaceID), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := c.doRequest(req, c.PrefectApiKey)
+	if err != nil {
+		return nil, err
+	}
+
+	blockDocuments := []BlockDocument{}
+
+	err = json.Unmarshal(body, &blockDocuments)
+	if err != nil {
+		return nil, err
+	}
+
+	return blockDocuments, nil
+}
+
+func (c *Client) GetBlockDocument(blockDocumentId string, workspaceID string) (*BlockDocument, error) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/accounts/%s/workspaces/%s/block_documents/%s", c.PrefectApiUrl, c.PrefectAccountId, workspaceID, blockDocumentId), nil)
```

Comment:
We can refactor later, but a lot of our client code in Python includes the `accounts` and `workspaces` path parts as part of the URL field, so that we can support both Prefect Server and Prefect Cloud, so that may be something to consider here as well.

For example, rather than defining things this way, we could define `c.PrefectApiUrl` as `https://api.prefect.cloud/api/accounts/741fcb04-d564-11ed-afa1-0242ac120002/workspaces/7b916690-d564-11ed-afa1-0242ac120002` and then the code would only need to append `block_documents`. We would need other changes as well to better support Server (we would want to restrict resources that are not currently compatible with Server)

 Feel free to leave this as-is and we can log an issue to revise later.

Reply:
Hey @jawnsy I could refactor this one but I'd rather leave this with your team. You will know more about the differences and similarities between `Prefect Server` and `Prefect Cloud`.

One thing to note is that `workspaces` are created as part of the automation, so while we can set the terraform variable  up to the "account" level we'd still need to make it work to append the workspace in question before calling the `prefect_api` method. Think on a case where the automation sets 3 different workspaces, would you update the client on the fly or pass the URL to every function in the `prefect_api` side and not have it set in the client..

I hope it makes sense.

Reply:
Yup, that makes sense! Thank you :)